endpoint: remove revision check around L4 policy calculation

ianvernon opened the pull request:
Endpoint policy regeneration logic would check if the policy repository revision
changed from the endpoint's prior policy calculation to determine if the
endpoint's L4 policy should be calculated. However, endpoint L4 policy should be
calculated for a variety of other reasons, including endpoint identity change.
Thus, short-circuiting endpoint L4 policy regeneration only based of the
difference of the endpoint's policy revision and the policy repository's
revision is incorrect and could lead to cases where L4 policy for an endpoint is
not calculated.

Signed-off by: Ian Vernon <ian@cilium.io>

```release-note
Always recalculate endpoint L4 policy when regeneratePolicy is called
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5515)
<!-- Reviewable:end -->

Commit:
endpoint: remove revision check around L4 policy calculation

Endpoint policy regeneration logic would check if the policy repository revision
changed from the endpoint's prior policy calculation to determine if the
endpoint's L4 policy should be calculated. However, endpoint L4 policy should be
calculated for a variety of other reasons, including endpoint identity change.
Thus, short-circuiting endpoint L4 policy regeneration only based of the
difference of the endpoint's policy revision and the policy repository's
revision is incorrect and could lead to cases where L4 policy for an endpoint is
not calculated.

Signed-off by: Ian Vernon <ian@cilium.io>

aanm reviewed (dismissed):
can you add a CI test with your steps from https://github.com/cilium/cilium/pull/5515#issuecomment-420434437

Commit:
test: add CI test for endpoint with already-allocated identity

Add a CI test which ensures that policy is correctly computed for endpoints
which have an identity which was created in the key-value store prior to the
creation of said endpoint.

Signed-off by: Ian Vernon <ian@cilium.io>
